recipes/openai_ner.py

```diff
@@ -304,20 +149,25 @@ def ner_openai_correct(
     max_examples: int = 2,
     verbose: bool = False,
 ):
-    examples = _read_prompt_examples(examples_path)
+    examples = read_prompt_examples(examples_path, example_class=NERPromptExample)
     nlp = spacy.blank(lang)
     if segment:
         nlp.add_pipe("sentencizer")
-    api_key, api_org = _get_api_credentials(model)
+    api_key, api_org = get_api_credentials(model)
     openai = OpenAISuggester(
+        response_parser=make_ner_response_parser(labels=labels, lang=lang),
         openai_model=model,
         labels=labels,
         max_examples=max_examples,
-        prompt_template=_load_template(prompt_path),
+        prompt_template=load_template(prompt_path),
         segment=segment,
         verbose=verbose,
         openai_api_org=api_org,
         openai_api_key=api_key,
+        openai_n=1,
+        openai_retry_timeout_s=10,
+        openai_read_timeout_s=20,
+        openai_n_retries=10,
```

Comment:
The `OpenAISuggester` now accepts a `response_parser: Callable` that defines how to create a Prodigy annotation task example given ChatGPT's response and some current info in a given example dict. The `make_ner_response_parser` is a factory that creates this function.